postgres/postgres.go

```diff
@@ -16,45 +16,80 @@ type Options struct {
 	User     string
 	Password string
 	Database string
+	Table    string
 }
 
+// Queries are the postgres queries for a given table.
 type Queries struct {
+	deleteQuery  string
+	existsQuery  string
+	getQuery     string
+	putQuery     string
+	queryQuery   string
+	prefixQuery  string
+	limitQuery   string
+	offsetQuery  string
+	getSizeQuery string
 }
 
-func (Queries) Delete() string {
-	return `DELETE FROM blocks WHERE key = $1`
+// NewQueries creates a new PostgreSQL set of queries for the passed table
+func NewQueries(tbl string) Queries {
+	return Queries{
+		deleteQuery:  fmt.Sprintf("DELETE FROM %s WHERE key = $1", tbl),
+		existsQuery:  fmt.Sprintf("SELECT exists(SELECT 1 FROM %s WHERE key=$1)", tbl),
+		getQuery:     fmt.Sprintf("SELECT data FROM %s WHERE key = $1", tbl),
+		putQuery:     fmt.Sprintf("INSERT INTO %s (key, data) VALUES ($1, $2) ON CONFLICT (key) DO UPDATE SET data = $2", tbl),
```

Comment:
this is the only thing i see here that i think is postgres specific, right? sadly there's not a great standard for this - mysql/sqlite both use `Replace into` instead. is there a way to communicate / allow this to work with other sql backends?

Reply:
Yes I think so. It would be nice to just provide a Queries object for some big DB providers but we’d need to add tests for each one. Future PR maybe?

Reply:
yeah, seems reasonable to not solve this entirely now.